File: testdata/fixtures/java-service/src/main/java/com/acme/pricing/Money.java
package com.acme.pricing;

public record Money(long cents) {
    public static final Money ZERO = new Money(0);

    public Money plus(Money other) {
        return new Money(cents + other.cents());
    }

    public Money percent(int percent) {
        return new Money(cents * percent / 100);
    }
}

File: testdata/fixtures/py-service/app/pricing/money.py
from dataclasses import dataclass


@dataclass
class Money:
    cents: int = 0

    def plus(self, other: "Money") -> "Money":
        return Money(self.cents + other.cents)

    def percent(self, p: int) -> "Money":
        return Money(self.cents * p // 100)


ZERO = Money()
